native/src/components/Header.tsx

```diff
@@ -218,13 +220,30 @@ const Header = ({
     }
 
     const previousRouteTitle = (previousParams as { title?: string } | undefined)?.title
-    return previousRouteTitle ?? t(previousRoute.name === 'categories' ? 'localInformation' : previousRoute.name)
+    return previousRouteTitle ?? t(previousRoute.name === CATEGORIES_ROUTE ? 'localInformation' : previousRoute.name)
+  }
+
+  const isHeaderTextInPhoneLanguage = (): boolean => {
```

Comment:
Hmm... not sure if I completely understand this function. I also think its not entirely correct, e.g. if following an internal link, right?
How about: `getHeaderText()` returns both a text and a boolean indicating the language it is in? e.g. `{ text: previousRouteTitle, systemLanguage: false }` and `{ text: t('locations'), systemLanguage: true }`. Should be less complexity and less error prone without duplicated logic/if else.

Reply:
Hmm, that's an interesting idea. I also realised that I was going on the possibly incorrect idea that if the route back is the name of the region (e.g. Testumgebung), it should be in the language of the app, not the one of the system. But now I'm not so sure, it might make more sense to use the system language. What do you think?

On a side note, I checked, we don't translate region names (e.g. Munich is München in all languages)

Reply:
If it is not translated anyway, I don't see a reason why I would prefer one over the other. We can also hardcode it to `German` if its the city name.